src/main/java/frc/robot/Robot.java

```diff
@@ -115,6 +119,18 @@ public void robotInit() {
     public void robotPeriodic() {
         CommandScheduler.getInstance().run();
         VirtualSubsystem.run();
+
+        final ShotParameters.Parameters shotParameters = ShotParameters.get(
+                robotContainer.swerve.getPose()
+                        .minus(FieldConstants.getSpeakerPose())
+                        .getTranslation()
+                        .getNorm()
+        );
+
+        Logger.recordOutput("ShotParameters/ArmPivotAngle", shotParameters.armPivotAngle());
+        Logger.recordOutput("ShotParameters/LeftVelocityRotsPerSec", shotParameters.leftVelocityRotsPerSec());
+        Logger.recordOutput("ShotParameters/RightVelocityRotsPerSec", shotParameters.rightVelocityRotsPerSec());
+        Logger.recordOutput("ShotParameters/AmpVelocityRotsPerSec", shotParameters.ampVelocityRotsPerSec());
```

Comment:
This `ShotParameters` stuff is probably a little messy (not sure if it should live here), but I think it can be merged in for now and we can address this later.